schema/schema.go

```diff
@@ -0,0 +1,70 @@
+package schema
+
+import (
+	"os"
+	"strings"
+
+	"github.com/sketch-hq/gql-lint/introspection"
+	"github.com/sketch-hq/gql-lint/parser"
+	"github.com/vektah/gqlparser/v2/ast"
+	gqlvalidator "github.com/vektah/gqlparser/v2/validator"
+)
+
+const deprecated = `directive @deprecated(
+	reason: String = "No longer supported"
+) on FIELD_DEFINITION | ARGUMENT_DEFINITION | ENUM_VALUE | INPUT_FIELD_DEFINITION`
+
+func Load(source string) (*ast.Schema, error) {
+	loader := schemaSources(source)
+	sources, err := loader.Load(source)
+	if err != nil {
+		return nil, err
+	}
+
+	return parser.ParseSchema(sources...)
+}
+
+type SchemaSource interface {
+	Load(source string) ([]*ast.Source, error)
+}
+
+type FileSchemaSource struct{}
+
+func (s FileSchemaSource) Load(source string) ([]*ast.Source, error) {
+	contents, err := os.ReadFile(source)
+	if err != nil {
+		return nil, err
+	}
+
+	return []*ast.Source{
+		gqlvalidator.Prelude,
+		{Name: source, Input: string(contents), BuiltIn: false},
+	}, nil
+}
+
+type HttpSchemaSource struct{}
+
+func (s HttpSchemaSource) Load(source string) ([]*ast.Source, error) {
+	contents, err := introspection.Load(source)
+	if err != nil {
+		return nil, err
+	}
+
+	schemaString := string(contents)
+	sources := []*ast.Source{{Name: source, Input: schemaString, BuiltIn: true}}
+
+	// workaround for absinthe based graphql servers that does NOT include the deprecated
+	// directive in the schema.
+	if !strings.Contains(schemaString, "directive @deprecated") {
+		sources = append(sources, &ast.Source{Input: deprecated, BuiltIn: false})
+	}
+
+	return sources, nil
+}
+
+func schemaSources(source string) SchemaSource {
```

Comment:
`loaderForSource` maybe? wdyt?

Reply:
Yup, that sounds better. Especially after renaming the interface to `Loader`.